Userland/Services/WindowServer/Menubar.h

```diff
@@ -49,8 +50,7 @@ class Menubar
     void for_each_menu(Callback callback)
     {
         for (auto& menu : m_menus) {
-            if (callback(*menu) == IterationDecision::Break)
-                return;
+            DO_CALLBACK(callback, *menu);
```

Comment:
the way it is now you don't break out; try
```cpp
if (do_callback(callback, *menu))
    break;
```
instead (assuming the implementation from @alimpfard)

Reply:
Thank you, that is what the implementation looks like now.